pkg/util/http.go

```diff
@@ -66,3 +66,11 @@ func HexStringToTraceID(id string) ([]byte, error) {
 
 	return byteID, nil
 }
+
+// TraceIDToHexString converts a trace ID to its string representation and removes any leading zeros.
```

Comment:
any particular reason to trim 0s? We just use `hex.EncodeToString()` throughout the codebase to go from bytes => string for trace ids

Reply:
@kvrhdn Thoughts?

Reply:
Just adding a thought:  It might be to trim the leading zeroes from a 8-byte trace ID stored in a 16-byte slice, to make the trace ID simpler.

Reply:
I added this function because it mirrors nicely with `HexStringToTraceID`. It encapsulates how we stringify trace IDs which is nice for people newer to the codebase. It's not immediately obvious that `hex.EncodeToString()` is the best way to print a trace ID so they'll have to search around for other places where we do this (at least that's what I had to do).
Even better would be to have a `TraceID` type with some associated functions to convert it (I think OTel has this), though I'm not sure if this is worth it as this will add some boilerplate.

Regarding trim 0s: I added this to avoid seeing a ton of zeroes when you are working with 64-bit trace IDs. I believe this is one of the first places where we show trace IDs as output to users, so I thought this would be a bit friendlier. Though we could also argue Grafana should do this for us and not bother.
The current implementation is a bit rough though... a trace ID `0` will be converted into an empty string 🙃 A better implementation would be to attempt to detect whether our trace ID is 64-bit and if so, only encode the last 8 bytes.

Reply:
+1 to @kvrhdn
Agree with all points made.